### ISSUE TITLE
Is it possible to maintain the quality of the input image?

### ISSUE THREAD

**Issue by kristiannordestgaard** [EVA-01/pxlsrt#10]
Can't find an option for this. If it does not exist, can I tweak the code somehow, or could the feature maybe be added?

**Comment by czycha** [comment]
You're welcome to submit a pull request if you would like. Admittedly been a while since I've looked at this project or worked in Ruby at all. 

Is this for the jRuby or plain Ruby implementation? Ruby version only works with PNGs so it is surprising to hear that it is degrading the quality. Could be something wrong with oily_png if that's the case.

**Comment by kristiannordestgaard** [comment]
This should be plain Ruby, and it's called pxlsrt-1.8.2. 

My file size goes from 46 mB to 51 kB. This is the string I use:

`pxlsrt brute input.png output.png --min 100 --max 200 --vertical --smooth --reverse --method hue`

I will try the NPM version and see what happens ...

**Comment by czycha** [comment]
If you could, could you provide the input image you are using? That's a dramatic drop in filesize and I'm curious what is actually happening to the visuals of the image.

**Comment by kristiannordestgaard** [comment]
The input image is not meant for the public, but here is a different, fairly similar image, which also gives a dramatic decrease in file size. 

https://we.tl/t-K3xml5pqNN

**Comment by czycha** [comment]
The file size is shrinking significantly mainly because whatever method was used to originally save it was inefficient. I see no representative downgrades in quality such as artifacts or color formatting issues when passed through pxlsrt.

The "no-op" command for pxlsrt is the following:
```bash
pxlsrt brute IMG_20190902_012102-v2.png -o o.png --min 1 --max 1
```

This basically does no transformations since it is only sorting sets of pixels with length 1. Output file size goes from 47.9MB to 79KB despite no changes taking place and the image remaining the same visually.

Further proof is that if I run [image_optim](https://github.com/toy/image_optim) on the original image I have the following compression results:
<img width="393" alt="Screen Shot 2019-09-04 at 9 04 30 AM" src="https://user-images.githubusercontent.com/4816472/64261982-05ba9700-cef3-11e9-861f-42923001536e.png">

After nearly four minutes, it trimmed 99% of the fat from the file to reduce from 47.9MB to 22KB.

In short, whichever method was used to encode your image as PNG is less efficient than the one pxlsrt uses. I'm going to create a separate issue for changing the PNG encoding method which could introduce benefits to save speed at the cost of file size.

**Comment by kristiannordestgaard** [comment]
I see. I was not paying much attention when exporting the original image.
